examples/workflows/unet_training_dict.py

```diff
@@ -163,7 +163,7 @@ def main():
         optimizer=opt,
         loss_function=loss,
         inferer=SimpleInferer(),
-        amp=True,  # if no FP16 support in GPU or PyTorch version < 1.6, change to False
+        amp=True if monai.config.get_torch_version_tuple() >= (1, 6) else False,
```

Comment:
This does not test if the GPU supports FP16 as the comment previous demanded.

Reply:
Hi @hjmjohnson ,

Do you mean to keep the previous comment?
I added comments back in the latest commit.

Thanks.

Reply:
I expected some function that tested if the GPU available had FP16 capabilities, but a comment is the least acceptable option.

Reply:
Hi @hjmjohnson , please feel free to correct me if I misunderstand your question.
This is the example code for workflows, if the GPU has FP16 and PyTorch > 1.6, it will run in AMP mode.
And I also added the AMP integration test according to @wyli 's suggestion.
So why you think it's not tested?

Thanks.

Reply:
Is the introspective test for FP16 support somewhere else?

I was expecting 
```python
## psudeocode
COMPUTE_DEVICE=pytorch.device(0)
amp=True if ( monai.config.get_torch_version_tuple() >= (1, 6) \
                       and COMPUTE_DEVICE.is_gpu() \
                       and COMPUTE_DEVICE.supports_fp16() ) else False,
```

Reply:
Yes, this PR contains some integration test for AMP=True/False:
https://github.com/Project-MONAI/MONAI/pull/841/files#diff-f73d275529f0aa7b1d8f8bb82ecbe020R255

Thanks.